Source/Core/Core/IOS/Network/Socket.cpp

```diff
@@ -793,6 +811,20 @@ WiiSocket::ConnectingState WiiSocket::GetConnectingState()
   return connecting_state;
 }
 
+bool WiiSocket::IsTCP() const
+{
+  const auto state = Common::SaveNetworkErrorState();
+
+  int socket_type;
+  socklen_t option_length = sizeof(socket_type);
+  const bool is_tcp = getsockopt(fd, SOL_SOCKET, SO_TYPE, reinterpret_cast<char*>(&socket_type),
+                                 &option_length) == 0 &&
+                      socket_type == SOCK_STREAM;
+
+  Common::RestoreNetworkErrorState(state);
```

Comment:
Similar to my other comment, though less necessary here, if RAII was used for this, then you could save  variable and just return the `getsockopt` call.